scipy/optimize/_highspy/_highs_wrapper.py

```diff
@@ -336,3 +329,20 @@ def check_option(highs_inst, option, value):
     if status != _h.HighsStatus.kOk:
         return 4, "Failed to validate option value."
     return 0, "Check option succeeded."
+
+def _get_marg_bnds(solution_col_dual, basis_col_status):
+    numcol = len(basis_col_status)
+    marg_bnds = np.zeros((2, numcol))
+    basis_col_status_values = np.array(
+        [status.value for status in basis_col_status],
+        dtype='uint8'
+    )
+    solution_col_dual_array = np.array(solution_col_dual, dtype='float64')
+    is_lower = basis_col_status_values == _h.HighsBasisStatus.kLower.value
+    is_upper = basis_col_status_values == _h.HighsBasisStatus.kUpper.value
+    # If is_lower, write to row 0. If is_upper, write to row 1. If neither, write
+    # nothing. Note that is_lower and is_upper cannot be True at the same index.
+    marg_bnds = np.zeros((2, numcol))
```

Comment:
Isn't this duplicated from above?

> a speedup for linprog() of 0.83%

I won't speak for the `optimize` folks, but normally I'd be looking for a fair bit more than that on the perf side before a refactoring and the associated review/churn would be worth it. Maybe the extra testing has some merit I suppose.

Reply:
>Isn't this duplicated from above?

Sorry, I'm not following. What's duplicated?

Reply:
Check lines 335 and 345 :)

Reply:
Ah, thanks. Fixed in 95af3f7.